src/scripts/charmc

```diff
@@ -2090,21 +2081,21 @@ fi
 
 Debug "About to link..."
 
-if [ "$INPUT_GIVEN" != "1" ]
+if [[ "$INPUT_GIVEN" != "1" ]]
 then
     Abort "Trying to link, but no object files or library archives were specified"
 fi
 
 # if interoperate dont create the library now, wait to collect all info
-if [ "$MPI_INTEROPERATE" = "" ]
+if [[ "$MPI_INTEROPERATE" = "" ]]
 then
 case "$OBJECT" in
 *.o)
 	Debugf "Converting $FILES to $OBJECTFILES"
 	nFiles=`echo $FILES | wc -w`
 	nObj=`echo $OBJECTFILES | wc -w`
 # One source file, one .o: just move the .o
-	if [ $OBJECT != $OBJECTFILES ]
+	if [[ " $OBJECT" != "$OBJECTFILES" ]]
```

Comment:
```suggestion
	if [[ "$OBJECT" != "$OBJECTFILES" ]]
```

Reply:
That won't work, since $OBJECTFILES has a leading space in the single file case.

Reply:
Something is not right here, because the `mv` command in the `then` clause of this `if` statement is what is failing on Darwin builds. In any case, could this be noted in a comment?